### PR TITLE
[Connect] Don't preview downloads + better error handling

### PR DESCRIPTION
**Pull request by mludowise-stripe**
## Summary
- Directly open system share sheet instead of previewing downloaded file
- If the file doesn't exist after it should have downloaded, displays an error message to the user
- Deletes the temp file after the preview or action sheet is closed
- Stubs out some TODOs to add error logging for various edge cases

## Motivation
https://jira.corp.stripe.com/browse/MXMOBILE-2827

## Testing
- [x] Manually tested (see video)
- [x] Added unit test coverage

https://github.com/user-attachments/assets/520079a0-4840-4d06-861b-a86bb7646afd

### CODE REVIEW

File: StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift [hunk 1]
```diff
@@ -111,16 +116,31 @@ private extension ConnectWebView {
...
             message: NSError.stp_unexpectedErrorMessage(),
             preferredStyle: .alert)
         presentPopup(alert)
     }
+
+    func cleanupDownloadedFile() {
+        guard let downloadedFile else { return }
+
+        // Delete file since we don't need it anymore
+        DispatchQueue.global(qos: .background).async { [weak self] in
```
[the remark this answers is not in the document]
Comment (mludowise-stripe) [reply StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift:134]: I'm not entirely sure we _need_ to dispatch, but go/llm indicates that it's best practice since it's synchronous and can take some time. But Apple's docs don't state a recommendation:
https://developer.apple.com/documentation/foundation/filemanager/1413590-removeitem

[the remark this answers is not in the document]
Comment (mludowise-stripe) [reply StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift:134]: Related - should we dispatch when creating the temp directory? I don't think so given that `decideDestinationUsing` is async, but interested in feedback
https://github.com/stripe/stripe-ios/blob/master/StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift#L246

**Comment by mludowise-stripe** [comment]
Putting this back into draft in light of feedback from bug bash. Going to pivot to just showing share sheet instead of preview

**Comment by mludowise-stripe** [comment]
Skipping dead code check – method is being called from a delegate callback

File: StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift [hunk 2]
```diff
@@ -259,12 +280,23 @@ extension ConnectWebView {
...
+            cleanupDownloadedFile()
+            return
+        }
 
-        // Display a preview of the file to the user
-        let previewController = QLPreviewController()
-        previewController.dataSource = self
-        previewController.modalPresentationStyle = .pageSheet
-        presentPopup(previewController)
+        // File is not previewable, so directly open share sheet instead
```
Comment (nschris-stripe) [inline StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift:294]: This comment needs to be updated since we don't show a preview ever now.

File: StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (nschris-stripe) [reply StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift:128]: This looks right to me.

**Review by nschris-stripe — DISMISSED** [review]

File: StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift [hunk 3]
```diff
@@ -111,16 +110,31 @@ private extension ConnectWebView {
...
             preferredStyle: .alert)
         presentPopup(alert)
     }
+
+    func cleanupDownloadedFile() {
+        guard let downloadedFile else { return }
+
+        // Delete file since we don't need it anymore
+        DispatchQueue.global(qos: .background).async { [weak self] in
+            try? self?.fileManager.removeItem(at: downloadedFile)
```
[the remark this answers is not in the document]
Comment (mludowise-stripe) [reply StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift:129]: Oh, I remember why I didn't do that now. I'm not well versed enough in Swift 6 language mode issues to know what the right fix is for this, so lmk if you have any suggestions. Otherwise I think the best option is to leave it as-is.

image

**Review by nschris-stripe — APPROVED** [review]
